src/otx/core/data/adapter/base_dataset_adapter.py

```diff
@@ -272,11 +272,16 @@ def _prepare_label_information(
 
         return {"category_items": category_items, "label_groups": label_groups, "label_entities": label_entities}
 
-    def _is_normal_polygon(self, annotation: DatumAnnotationType.polygon) -> bool:
+    def _is_normal_polygon(self, annotation: DatumAnnotationType.polygon, width: int, height: int) -> bool:
         """To filter out the abnormal polygon."""
         x_points = [annotation.points[i] for i in range(0, len(annotation.points), 2)]
         y_points = [annotation.points[i + 1] for i in range(0, len(annotation.points), 2)]
-        return min(x_points) < max(x_points) and min(y_points) < max(y_points)
+        return (
+            min(x_points) < max(x_points)
+            and min(y_points) < max(y_points)
+            and max(x_points) < width
+            and max(y_points) < height
+        )
```

Comment:
Can the changed code conditions solve the OTXDetDataset: Invalid rectangle with coordinates: x1=1.0, y1=0.14171337069655723, x2=1.0, y2=0.19495596477181745 issue? Isn't this condition still False in min(x_points) < max(x_points)?

Reply:
@chuneuny-emily The issue of invalidation occurs within the normalization in `_get_polygon_entity`. This occurs when the x or y coordinates exceed the dimensions of the image (e.g., image dimensions of 3500 x 3500, but points with coordinates such as (7000, 7000)). Consequently, this leads to the presence of invalid points in the following code:

```python
points=[
    Point(x=annotation.points[i] / width, y=annotation.points[i + 1] / height)
    for i in range(0, len(annotation.points), 2)
]
```

Reply:
@chuneuny-emily I updated the condition for readability and added a area check.